### DISCUSSION TITLE
JWT OAuth2 authentication

### DISCUSSION THREAD

**Discussion by zhaohanqing95** [grafana/grafana-infinity-datasource#272] [General]
hello, I would like to use JWT OAuth2 method to authenticate my api, but I have to use account/paaswd to get token from one api, then use the token to  authenticate the other api.
I found the JWT OAuth Type need Private Key and Token URL,How can I get JWT authentication by account and passwd?

**Comment by yesoreyeram** [comment]
Can you share the api document if any?

**Reply by zhaohanqing95** [reply]
https://api.clooops.com/docs/v1/ 
here the api document.

**Reply by zhaohanqing95** [reply]
I need to post the /login url api first, the result is the authentication token.
then use this token for all other REST API, But this token will expire after one hour.

**Comment by thegizzardwizard** [comment]
Has there been any progress on this? I need to pull data from multiple IoT sites and they are all using a user name and password scheme rather than a 'private key'. Would love to be able to access this data. Thanks for all the work on this extremely useful plugin.
